ocaml/xapi/xapi_vbd_helpers.ml

```diff
@@ -169,12 +169,12 @@ let valid_operations ~expensive_sharing_checks ~__context record _ref' : table =
     if not vdi_record.Db_actions.vDI_managed
     then set_errors Api_errors.vdi_not_managed [ _ref ] all_ops;
 
-    let vdi_operations_besides_copy =
+    let operation_cannot_be_performed_live=
```

Comment:
The nested negations in the code make my head spin. How about making it simpler and more efficient by avoiding `List.mem` (or similar):




```

let operation_cannot_be_performed_live =
  let live_op = function
  | `snapshot
  | `resize_online
  | `blocked
  | `clone
  | `mirror
  | `copy -> true
  | _     -> false
  in
  List.filter live_op vdi_record.Db_actions.vDI_current_operations
```